### DISCUSSION TITLE
Full restart

### DISCUSSION THREAD

**Discussion by FieldingChase** [Nek5000/nekRS#603] [General]
Is there an NekRS analogue of Nek5000's full_restart subroutine?

**Comment by stgeke** [comment]
No but you could do this through the nek5000 interface. What are you after which requires an accurate restart?

**Comment by FieldingChase** [comment]
I am doing time-averaging of a turbulent channel flow for turbulence statistics. When I was running this simulation in Nek5000 on a cluster, the simulation would hit the wall-time and stop before it had run for as long as I intended. I did the full restart to affect the averaging over time as little as possible when I averaged both runs together. I might not need this full restart when I run NekRS on GPUs. Anyway, I will take a look at doing this full restart through the Nek5000 interface. 

Thanks for your help.

**Reply by stgeke** [reply]
I don't think for this application you would need an accurate restart.

**Reply by stgeke** [reply]
btw: if you specify in par

```
checkpointInterval = 0 # dump field file at the end only
stopAt = elapsedTime
elapsedTime = 55 # 5 min smaller than job time to account for time spent during setup
```

you can ensure you'll get a checkpoint file at the end of your run.

**Reply by FieldingChase** [reply]
Okay, thanks for your assistance.
